### PR TITLE
CI: temporarily drop numpy to under 2 in pypi builds

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Force the pypi builds to run with numpy<2

## Motivation and Context

Some of these builds are failing to no fault of their source PRs.

## How Has This Been Tested?

It worked on pcdsdevices

## Where Has This Been Documented?

Here only

## Pre-merge checklist
- [x] Code works interactively (a real hutch config file can be loaded)
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran ``docs/pre-release-notes.sh`` and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
There isn't currently a spot in the .github config to set package overrides for the docs build, so I have put this into the docs requirements instead.

**Comment by tangkong** [comment]
The most recent np2 related failure I've seen is a `pcdscalc` -> `xraylib` incompatibility.  We're going to need xraylib to update before this will ever work, and that package is unfortunately out of our purview.

File: docs-requirements.txt [hunk 1]
```diff
@@ -4,3 +4,5 @@ sphinx
 sphinx_rtd_theme
 sphinxcontrib-jquery
 pypandoc
+# Temporary pin, can remove once it installs without this
```
Comment (tangkong) [inline docs-requirements.txt:7]:
```suggestion
# Temporary pin, xraylib 4.1.3 (pcdscalc) incompatible
```
I think future me will need more verbosity here

**Review by tangkong — COMMENTED** [review]
Maybe we just pin this upstream in the ci helpers.  But recording the incompatibility here is probably also appropriate.  We can pin in both places

**Review by tangkong — APPROVED** [review]

**Pull request merged**